### ISSUE TITLE
 Unhandled Exception: MissingPluginException(No implementation found for method initialize on channel dexterous.com/flutter/local_notifications)

### ISSUE THREAD

**Issue by Dominic12** [MaikuB/flutter_local_notifications#238]
I'm trying to fire a notification when the background location callback returns. But this leads into the following exception:  Unhandled Exception: MissingPluginException(No implementation found for method initialize on channel dexterous.com/flutter/local_notifications)

I know, that there was an issue with the same problem before, but there was no solution provided.

**Comment by MaikuB** [comment]
Yep there was a similar issue reported before but couldn't be reproduced. This kind of error occurs when the plugin can't find the method in the native side, in which case that would mean the plugin doesn't work at all but that's not actually the case. Without a code sample, it's difficult to know how you've run into that situation. There's a similar issue on one of the official plugins where one of the Flutter team has essentially posted a checklist of things to go through that may help in your case [here](https://github.com/flutter/flutter/issues/10912#issuecomment-310499594).

Plugins like those run code in the background and can execute code without the app generally require extra setup when said code needs to call functions belonging in another plugin. I'm not sure if that has been done in your case

**Comment by Dominic12** [comment]
Hi, i could not examine if it fails on both platforms but i can give you my native files.


```
App Delegate.m

#include "AppDelegate.h"
#include "GeneratedPluginRegistrant.h"
#import "GoogleMaps/GoogleMaps.h"
#include "GeneratedPluginRegistrant.h"
// Add the import for the GeofencingPlugin.



@implementation AppDelegate

- (BOOL)application:(UIApplication *)application
    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
  [GMSServices provideAPIKey:@"key"];
  [GeneratedPluginRegistrant registerWithRegistry:self];
  // Override point for customization after application launch.
    
 
    
    // Note: if we return NO, this vetos the launch of the application.
  return [super application:application didFinishLaunchingWithOptions:launchOptions];
}

@end

```

GeneratePluginRegistrant.m

//
//  Generated file. Do not edit.
//

#import "GeneratedPluginRegistrant.h"
#import <android_intent/AndroidIntentPlugin.h>
#import <cloud_firestore/CloudFirestorePlugin.h>
#import <cloud_functions/CloudFunctionsPlugin.h>
#import <firebase_auth/FirebaseAuthPlugin.h>
#import <firebase_core/FirebaseCorePlugin.h>
#import <flutter_local_notifications/FlutterLocalNotificationsPlugin.h>
#import <fluttertoast/FluttertoastPlugin.h>
#import <geolocator/GeolocatorPlugin.h>
#import <google_api_availability/GoogleApiAvailabilityPlugin.h>
#import <google_maps_flutter/GoogleMapsPlugin.h>
#import <location_background_plugin/LocationBackgroundPlugin.h>
#import <permission_handler/PermissionHandlerPlugin.h>
#import <uni_links/UniLinksPlugin.h>
#import <vibrate/VibratePlugin.h>

@implementation GeneratedPluginRegistrant

+ (void)registerWithRegistry:(NSObject<FlutterPluginRegistry>*)registry {
  [FLTAndroidIntentPlugin registerWithRegistrar:[registry registrarForPlugin:@"FLTAndroidIntentPlugin"]];
  [FLTCloudFirestorePlugin registerWithRegistrar:[registry registrarForPlugin:@"FLTCloudFirestorePlugin"]];
  [CloudFunctionsPlugin registerWithRegistrar:[registry registrarForPlugin:@"CloudFunctionsPlugin"]];
  [FLTFirebaseAuthPlugin registerWithRegistrar:[registry registrarForPlugin:@"FLTFirebaseAuthPlugin"]];
  [FLTFirebaseCorePlugin registerWithRegistrar:[registry registrarForPlugin:@"FLTFirebaseCorePlugin"]];
  [FlutterLocalNotificationsPlugin registerWithRegistrar:[registry registrarForPlugin:@"FlutterLocalNotificationsPlugin"]];
  [FluttertoastPlugin registerWithRegistrar:[registry registrarForPlugin:@"FluttertoastPlugin"]];
  [GeolocatorPlugin registerWithRegistrar:[registry registrarForPlugin:@"GeolocatorPlugin"]];
  [GoogleApiAvailabilityPlugin registerWithRegistrar:[registry registrarForPlugin:@"GoogleApiAvailabilityPlugin"]];
  [FLTGoogleMapsPlugin registerWithRegistrar:[registry registrarForPlugin:@"FLTGoogleMapsPlugin"]];
  [LocationBackgroundPlugin registerWithRegistrar:[registry registrarForPlugin:@"LocationBackgroundPlugin"]];
  [PermissionHandlerPlugin registerWithRegistrar:[registry registrarForPlugin:@"PermissionHandlerPlugin"]];
  [UniLinksPlugin registerWithRegistrar:[registry registrarForPlugin:@"UniLinksPlugin"]];
  [VibratePlugin registerWithRegistrar:[registry registrarForPlugin:@"VibratePlugin"]];
}

@end

**Comment by Dominic12** [comment]
i just noticed, that the initialisation works fine. The error throws when im trying to call the .show method.

**Comment by MaikuB** [comment]
You've mentioned intialisation works fine now so something seems to have changed there. Exactly what error are you seeing when calling show?

I would really need a complete sample app to reproduce the problem. The little bits of code don't actually show how it all hangs together to know what's wrong. It could well be that you've actually run into the same problem that I've provided a solution for in [this issue](https://github.com/MaikuB/flutter_local_notifications/issues/234#issuecomment-493073152)

**Comment by Dominic12** [comment]
Hi, thanks for responding so fast!

Yes, it seems like i have the exact same issue. But i cannot solve it somehow. I have moved all the methods out of the wrapper class and they are now directly in the main file.

```
import 'package:flutter/material.dart';
import 'package:flutter_localizations/flutter_localizations.dart';
import 'package:sbb_hunting_frontend/sbbtheme/sbb.dart';
import 'package:sbb_hunting_frontend/treasureHuntSelection.dart';
import 'package:sbb_hunting_frontend/i18n/SbbHuntingLocalizationsDelegate.dart';
import 'package:sbb_hunting_frontend/services/authentication.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:location_background_plugin/location_background_plugin.dart';
import 'dart:isolate';
import 'package:flutter_local_notifications/flutter_local_notifications.dart';


final locationManager = LocationBackgroundPlugin();

FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin;

  Future onDidReceiveLocalNotification(
      int id, String title, String body, String payload) async {
    // display a dialog with the notification details, tap ok to go to another page
  }
  Future onSelectNotification(String payload) async {
    if (payload != null) {
      print('notification payload: ' + payload);
    }
  }



  Future<void> showNotification(
      String notificationTitle, String notificatoinBody) async {
    var androidPlatformChannelSpecifics = AndroidNotificationDetails(
        'your channel id', 'your channel name', 'your channel description',
        importance: Importance.Max, priority: Priority.High, ticker: 'ticker');
    var iOSPlatformChannelSpecifics = IOSNotificationDetails();
    var platformChannelSpecifics = NotificationDetails(
        androidPlatformChannelSpecifics, iOSPlatformChannelSpecifics);
    await flutterLocalNotificationsPlugin.show(
        0, 'plain title', 'plain body', platformChannelSpecifics,
        payload: 'item x');
  }

void locationUpdateCallback(Location location) {
  print('Location Update: $location');
      flutterLocalNotificationsPlugin = new FlutterLocalNotificationsPlugin();

  showNotification("title", location.toJson());
}

Future<void> startMonitoringLocationChanges(){
      locationManager.monitorSignificantLocationChanges(locationUpdateCallback);

}
  
    
Future<void> stopMonitoringLocationChanges() =>
    locationManager.cancelLocationUpdates();

@JsonSerializable()
void main() => runApp(App());

class App extends StatelessWidget {
  LocationBackgroundPlugin _locationPlugin;
  Location _lastLocation;
  bool _isTracking = false;

  @override

  Widget build(BuildContext context) {
    var auth = new Auth();
    // auth.signOut();
    startMonitoringLocationChanges();

    return MaterialApp(
        title: 'SBB Hunting',
        theme: createSBBTheme(context),
        home: TreasureHuntSelection(auth: auth),
        // TreasureHuntSelection(),
        supportedLocales: [
          Locale('en'),
          Locale('de'),
          Locale('fr'),
          Locale('it')
        ],
        localizationsDelegates: [
          SbbHuntingLocalizationsDelegate(),
          GlobalMaterialLocalizations.delegate,
          GlobalWidgetsLocalizations.delegate
        ],
        localeResolutionCallback:
            (Locale locale, Iterable<Locale> supportedLocales) {
          if (supportedLocales == null || locale == null) {
            return null;
          }
          for (Locale supportedLocale in supportedLocales) {
            if (supportedLocale.languageCode == locale.languageCode ||
                supportedLocale.countryCode == locale.countryCode) {
              return supportedLocale;
            }
          }
          return supportedLocales.first;
        });
  }
}
```

Which throws me this error:

```
ocation Update: [2019-05-21 11:03:35.197Z] (37.33358124, -122.04152168) altitude: 5.0 m/s: 6.53
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: MissingPluginException(No implementation found for method show on channel dexterous.com/flutter/local_notifications)
#0      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:300:7)
<asynchronous suspension>
#1      FlutterLocalNotificationsPlugin.show (package:flutter_local_notifications/src/flutter_local_notifications.dart:120:20)
<asynchronous suspension>
#2      showNotification (package:sbb_hunting_frontend/main.dart:37:43)
<asynchronous suspension>
#3      locationUpdateCallback (package:sbb_hunting_frontend/main.dart:46:3)
#4      _backgroundCallbackDispatcher.<anonymous closure> (package:location_background_plugin/location_background_plugin.dart:101:22)
<asynchronous suspension>
#5      MethodChannel._handleAsMethodCall (package:flutter/src/services/platform_channel.dart:383:55)
<asynchronous suspension>
#6      MethodChannel.setMethodCallHandler.<anonymous closure> (package:flutter/src/services/platform_chann<…>
f
```


I dont have any other references to your plugin so this is the whole code which may be in relationship. Also its the only code which runs as a background service.

Again, thank you very much!

**Comment by Dominic12** [comment]
So, a little update:

i have put all the code related to the background task and to the notification plugin in one function.

```
void locationUpdateCallback(Location location) async{
  print('Location Update: $location');
  FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin =
      new FlutterLocalNotificationsPlugin();

  var androidPlatformChannelSpecifics = AndroidNotificationDetails(
      'your channel id', 'your channel name', 'your channel description',
      importance: Importance.Max, priority: Priority.High, ticker: 'ticker');
  var iOSPlatformChannelSpecifics = IOSNotificationDetails();
  var platformChannelSpecifics = NotificationDetails(
      androidPlatformChannelSpecifics, iOSPlatformChannelSpecifics);
  await flutterLocalNotificationsPlugin.show(
      0, 'title', location.toJson(), platformChannelSpecifics,
      payload: 'item x');

}
```

But still error:

```
Unhandled Exception: MissingPluginException(No implementation found for method show on channel dexterous.com/flutter/local_notifications)
#0      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:300:7)
<asynchronous suspension>
#1      FlutterLocalNotificationsPlugin.show (package:flutter_local_notifications/src/flutter_local_notifications.dart:120:20)
<asynchronous suspension>
#2      locationUpdateCallback (package:sbb_hunting_frontend/main.dart:54:41)
<asynchronous suspension>
#3      _backgroundCallbackDispatcher.<anonymous closure> (package:location_background_plugin/location_background_plugin.dart:101:22)
<asynchronous suspension>
#4      MethodChannel._handleAsMethodCall (package:flutter/src/services/platform_channel.dart:383:55)
<asynchronous suspension>
#5      MethodChannel.setMethodCallHandler.<anonymous closure> (package:flutter/src/services/platform_channel.dart:351:54)
#6      BinaryMessages.handlePlatformMessage (package:f<…>
```

**Comment by MaikuB** [comment]
Have you tried displaying a notification on its own i.e. without having tied with the background location update callback?

**Comment by Dominic12** [comment]
Jup, then it works just fine.

**Comment by MaikuB** [comment]
It you could produce a standalone sample to look into this further then I can try to see take a look. The other thing I would suggest trying is pulling in another plugin and call in one the methods belonging to the plugin to see if you run into the same issue

**Comment by Dominic12** [comment]
Okay, can i send you the project privatly? I dont think i should post it publicly.

**Comment by Dominic12** [comment]
Do you have a Gitlab account? If yes, i could authorize you there.

**Comment by MaikuB** [comment]
I have one now (same account name) but it sounds like you're looking to get me access to a private project, one that is potentially a commercial one as well (?). It'd be preferable if this was a basic sample that can reproduce the problem as opposed to your real project. For example, you could fork this repository that already has an example app and make modifications to it

**Comment by Najeeb8680** [comment]
I also have the same issue and it resolved by migrating app to androidX

**Comment by MaikuB** [comment]
@Dominic12 have you checked if your app has been migrated to AndroidX as per @Najeeb8680's reply?

**Comment by MaikuB** [comment]
Closing due to lack of response

**Comment by mamczas** [comment]
I have the same issue. 
Unhandled Exception: MissingPluginException(No implementation found for method show on channel dexterous.com/flutter/local_notifications)
My app has benn migrated to AndroidX.

**Comment by aviraj** [comment]
I have the same issue. My app is on AndroidX as well. I am basically trying to call "await _flutterLocalNotificationsPlugin.show()" in the onBackgroundMEssageHandler when it throws the exception. This works fine from onMessage

**Comment by sinabirdevrim** [comment]
@aviraj I have the same issue.

**Comment by altanseckin** [comment]
i have the same issue @aviraj . @MaikuB can you look into this ?

**Comment by MaikuB** [comment]
I imagine it would happen to any plugin that uses that callback and so would be more appropriate to ask on the repository where the `firebase_messaging`

**Comment by altanseckin** [comment]
But what should i do here ? I am stuck on this error for 2 days now. Do you have any suggesttion ?

**Comment by MaikuB** [comment]
I'm not sure what you're trying to ask since I mentioned asking on the repository that hosts the `firebase_messaging` plugin. Have you done that?

**Comment by aritzg** [comment]
Same issue here.

I am trying to use together flutter_local_notifications and background_fetch plugins.

Notification works just fine when the app is running but when the event is captured while the app is not running and treated as headless task it fails.

In my headless task, alike Dominic12 I both initialize ( try to) show the the notification but show method fails and prints this message

[ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: MissingPluginException(No implementation found for method show on channel dexterous.com/flutter/local_notifications)
02-17 15:08:42.653 20948 20999 E flutter : #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:154:7)

Did anyone overcome this problem?

Regards

**Comment by mcaubrey** [comment]
Same issue. I made a new Flutter app. I copied the packages from the example pubspec.yml file, and then copy and pasted the example project's main.dart file into my main.dart file. I followed the instructions in the Readme for iOS integration.

**Comment by MaikuB** [comment]
@mcaubrey can you provide a link to a repository containing a minimal sample app that reproduces the problem?

**Comment by MaikuB** [comment]
Another thing is I've heard of cases where others added packages whilst debugging and running into an exception then. If that's the scenario you have then note that hot reload isn't suppose to work in those scenarios as far as I know

**Comment by mcaubrey** [comment]
> @mcaubrey can you provide a link to a repository containing a minimal sample app that reproduces the problem?

Well, I mean, my sample app is basically the flutter_local_notfication package's example app (unless I copy/pasted the code wrong). 

However, I just tried `flutter clean` and a restart and that fixed it for me.

**Comment by sigurdurrafn** [comment]
Plugin registration issue?
https://stackoverflow.com/questions/60471697/missingpluginexceptionno-implementation-found-for-method-show-on-channel-dexter/60576958#60576958

**Comment by altanseckin** [comment]
@sigurdurrafn ' s solution work for my collage and me. Pls try this solution @mcaubrey @aritzg @aviraj @mamczas @Dominic12

**Comment by varadkulk** [comment]
For Kotlin: Application.kt

	package your,package.name

	import io.flutter.app.FlutterApplication
	import io.flutter.plugin.common.PluginRegistry
	import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback
	import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin
	import io.flutter.view.FlutterMain
	import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService
	import com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin

	class Application : FlutterApplication(), PluginRegistrantCallback {

		override fun onCreate() {
			super.onCreate()
			FlutterFirebaseMessagingService.setPluginRegistrant(this);
			FlutterMain.startInitialization(this)
		}

		override fun registerWith(registry: PluginRegistry?) {
			if (!registry!!.hasPlugin("io.flutter.plugins.firebasemessaging")) {
				FirebaseMessagingPlugin.registerWith(registry!!.registrarFor("io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin"));
			}
			if (!registry!!.hasPlugin("com.dexterous.flutterlocalnotifications")) {
				FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
			}
		}
	}

**Comment by whatamelon** [comment]
@varadkulk How can I use this with Java code?
anyone else?

**Comment by krunduev** [comment]
I had the same issue with background messages on android.
After adding 
`FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));`
all working fine.

I think it better to add to documentation.

**Comment by followthemoney1** [comment]
> For Kotlin: Application.kt
> 
> ```
> package your,package.name
> 
> import io.flutter.app.FlutterApplication
> import io.flutter.plugin.common.PluginRegistry
> import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback
> import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin
> import io.flutter.view.FlutterMain
> import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService
> import com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin
> 
> class Application : FlutterApplication(), PluginRegistrantCallback {
> 
> 	override fun onCreate() {
> 		super.onCreate()
> 		FlutterFirebaseMessagingService.setPluginRegistrant(this);
> 		FlutterMain.startInitialization(this)
> 	}
> 
> 	override fun registerWith(registry: PluginRegistry?) {
> 		FirebaseMessagingPlugin.registerWith(registry!!.registrarFor("io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin"));
> 		FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));		
> 	}
> }
> ```

you save many hours to me, thx

**Comment by michael-ottink** [comment]
I ch

> For Kotlin: Application.kt
> 
> ```
> package your,package.name
> 
> import io.flutter.app.FlutterApplication
> import io.flutter.plugin.common.PluginRegistry
> import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback
> import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin
> import io.flutter.view.FlutterMain
> import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService
> import com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin
> 
> class Application : FlutterApplication(), PluginRegistrantCallback {
> 
> 	override fun onCreate() {
> 		super.onCreate()
> 		FlutterFirebaseMessagingService.setPluginRegistrant(this);
> 		FlutterMain.startInitialization(this)
> 	}
> 
> 	override fun registerWith(registry: PluginRegistry?) {
> 		FirebaseMessagingPlugin.registerWith(registry!!.registrarFor("io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin"));
> 		FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));		
> 	}
> }
> ```

The background message still cannot find the plugin. Help what am I doing wong?

https://stackoverflow.com/questions/62820872/how-do-i-correctly-register-a-plugin-with-the-fcm-plugin

**Comment by verbuchselt** [comment]
> 
> 
> I had the same issue with background messages on android.
> After adding
> `FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));`
> all working fine.
> 
> I think it better to add to documentation.

This worked best for me, thanks a lot!

**Comment by SandroMaglione** [comment]
> For Kotlin: Application.kt
> 
> ```
> package your,package.name
> 
> import io.flutter.app.FlutterApplication
> import io.flutter.plugin.common.PluginRegistry
> import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback
> import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin
> import io.flutter.view.FlutterMain
> import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService
> import com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin
> 
> class Application : FlutterApplication(), PluginRegistrantCallback {
> 
> 	override fun onCreate() {
> 		super.onCreate()
> 		FlutterFirebaseMessagingService.setPluginRegistrant(this);
> 		FlutterMain.startInitialization(this)
> 	}
> 
> 	override fun registerWith(registry: PluginRegistry?) {
> 		FirebaseMessagingPlugin.registerWith(registry!!.registrarFor("io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin"));
> 		FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));		
> 	}
> }
> ```

It worked for me too! We should add this to the docs

**Comment by puntodamar** [comment]
> FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));

where do i add this code?

**Comment by thang510023** [comment]
> > FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
> 
> where do i add this code?

add to Application.kt in your android project directory: android/app/src/main/kotlin/com/project/Application.kt

override fun registerWith(registry: PluginRegistry) {
        FirebaseCloudMessagingPluginRegistrant.registerWith(registry);
        FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
    }

**Comment by Lawrene16** [comment]
> > > FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
> > 
> > 
> > where do i add this code?
> 
> add to Application.kt in your android project directory: android/app/src/main/kotlin/com/project/Application.kt
> 
> override fun registerWith(registry: PluginRegistry) {
> FirebaseCloudMessagingPluginRegistrant.registerWith(registry);
> FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
> }

What is the java equivalent please

**Comment by Lawrene16** [comment]
> > > FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
> > 
> > 
> > where do i add this code?
> 
> add to Application.kt in your android project directory: android/app/src/main/kotlin/com/project/Application.kt
> 
> override fun registerWith(registry: PluginRegistry) {
> FirebaseCloudMessagingPluginRegistrant.registerWith(registry);
> FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
> }

What is the java equivalent please

**Comment by varadkulk** [comment]
> > > > FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
> > > 
> > > 
> > > where do i add this code?
> > 
> > add to Application.kt in your android project directory: android/app/src/main/kotlin/com/project/Application.kt
> > 
> > override fun registerWith(registry: PluginRegistry) {
> > FirebaseCloudMessagingPluginRegistrant.registerWith(registry);
> > FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
> > }
> 
> What is the java equivalent please

https://stackoverflow.com/questions/60471697/missingpluginexceptionno-implementation-found-for-method-show-on-channel-dexter/60576958#60576958

**Comment by dgaedcke** [comment]
is this still required or was this made obsolete by plugin-registry v2?

**Comment by buildbro** [comment]
I had this issue after adding the package to pubspec and doing a hot restart. Stopping the app and rerunning `flutter run` helped

**Comment by varadkulk** [comment]
> is this still required or was this made obsolete by plugin-registry v2?

No its not required 
Looks like it was made obsolete by plugin-registry v2

**Comment by lionuncle** [comment]
> For Kotlin: Application.kt
> 
> ```
> package your,package.name
> 
> import io.flutter.app.FlutterApplication
> import io.flutter.plugin.common.PluginRegistry
> import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback
> import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin
> import io.flutter.view.FlutterMain
> import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService
> import com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin
> 
> class Application : FlutterApplication(), PluginRegistrantCallback {
> 
> 	override fun onCreate() {
> 		super.onCreate()
> 		FlutterFirebaseMessagingService.setPluginRegistrant(this);
> 		FlutterMain.startInitialization(this)
> 	}
> 
> 	override fun registerWith(registry: PluginRegistry?) {
> 		FirebaseMessagingPlugin.registerWith(registry!!.registrarFor("io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin"));
> 		FlutterLocalNotificationsPlugin.registerWith(registry!!.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));		
> 	}
> }
> ```
Worked for me!

**Comment by dmoses-xh** [comment]
For others that run into this issue, I was able to solve it by looking at the example app.  In my AppDelegate.swift I was missing the last`GeneratedPluginRegistrant.register(with: self)` before the return:

```swift
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    // This is required to make any communication available in the action isolate.
    FlutterLocalNotificationsPlugin.setPluginRegistrantCallback { (registry) in
      GeneratedPluginRegistrant.register(with: registry)
    }
    if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    }

    GeneratedPluginRegistrant.register(with: self)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
```

In case it changes in the future, here is the link to the example application: https://github.com/MaikuB/flutter_local_notifications/blob/master/flutter_local_notifications/example/ios/Runner/AppDelegate.swift
